contracts/root/predicates/ERC20Predicate.sol

```diff
@@ -0,0 +1,191 @@
+pragma solidity ^0.5.2;
+
+import { BytesLib } from "../../common/lib/BytesLib.sol";
+import { Common } from "../../common/lib/Common.sol";
+import { RLPEncode } from "../../common/lib/RLPEncode.sol";
+import { RLPReader } from "solidity-rlp/contracts/RLPReader.sol";
+import { IPredicate } from "./IPredicate.sol";
+import { Registry } from "../../common/Registry.sol";
+// import { WithdrawManager } from "../withdrawManager/WithdrawManager.sol";
+
+contract ERC20Predicate is IPredicate {
+  using RLPReader for bytes;
+  using RLPReader for RLPReader.RLPItem;
+
+  bytes32 constant DEPOSIT_EVENT_SIG = 0x4e2ca0515ed1aef1395f66b5303bb5d6f1bf9d61a353fa53f73f8ac9973fa9f6;
+  bytes32 constant WITHDRAW_EVENT_SIG = 0xebff2602b3f468259e1e99f613fed6691f3a6526effe6ef3e768ba7ae7a36c4f;
+  bytes32 constant LOG_TRANSFER_EVENT_SIG = 0xe6497e3ee548a3372136af2fcb0696db31fc6cf20260707645068bd3fe97f3c4;
+  // 0x2e1a7d4d = keccak256('withdraw(uint256)').slice(0, 4)
+  bytes4 constant WITHDRAW_FUNC_SIG = 0x2e1a7d4d;
+  // 0xa9059cbb = keccak256('transfer(address,uint256)').slice(0, 4)
+  bytes4 constant TRANSFER_FUNC_SIG = 0xa9059cbb;
+  bytes constant public networkId = "\x0d";
+
+  constructor(address _withdrawManager) public IPredicate(_withdrawManager) {}
+
+  /**
+   * @notice Start an exit from the side chain by referencing the preceding (reference) transaction
+   * @param data RLP encoded data of the reference tx that encodes the following fields:
+   * headerNumber Header block number of which the reference tx was a part of
+   * blockProof Proof that the block header (in the child chain) is a leaf in the submitted merkle root
+   * blockNumber Block number of which the reference tx is a part of
+   * blockTime Reference tx block time
+   * blocktxRoot Transactions root of block
+   * blockReceiptsRoot Receipts root of block
+   * receipt Receipt of the reference transaction
+   * receiptProof Merkle proof of the reference receipt
+   * branchMask Merkle proof branchMask for the receipt
+   * logIndex Log Index to read from the receipt
+   * exitTx Signed exit transaction
+   */
+  function startExit(bytes memory data)
```

Comment:
Can we use same interface for all predicates?

Reply:
Also calculate the cost for ERC20 exit.

Reply:
Yep, predicates will use the same interface `function startExit(bytes memory data) public;`.

> Also calculate the cost for ERC20 exit

Will do.